src/screens/AirdropDashboard/components/WalletBalanceCircle.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC } from 'react';
+import { View, StyleSheet, Text } from 'react-native';
+import { AnimatedCircularProgress } from 'react-native-circular-progress';
+
+import { typography, palette } from 'app/styles';
+
+const i18n = require('../../../../loc');
+
+interface Props {
+  balance: number;
+  threshold: number;
+}
+
+export const WalletBalanceCircle: FC<Props> = ({ balance = 0, threshold }: Props) => (
+  <AnimatedCircularProgress
+    size={149}
+    width={8}
+    fill={(balance / threshold) * 100}
+    tintColor={palette.textSecondary}
+    backgroundColor={palette.lightGrey}
+  >
+    {() => (
```

Comment:
why function here? what does it do?

Reply:
you can pass `fill` (percentage value) property back from the CircularComponent, although we don't need it so I omited it. Tried to straight pass React element, but it doesn't work like that, have to be function.